src/server/activeobjectmgr.cpp

```diff
@@ -97,15 +101,17 @@ void ActiveObjectMgr::removeObject(u16 id)
 {
 	verbosestream << "Server::ActiveObjectMgr::removeObject(): "
 			<< "id=" << id << std::endl;
-	ServerActiveObject *obj = getActiveObject(id);
-	if (!obj) {
+	auto it = m_active_objects.find(id);
+	if (it == m_active_objects.end()) {
 		infostream << "Server::ActiveObjectMgr::removeObject(): "
 				<< "id=" << id << " not found" << std::endl;
 		return;
 	}
 
-	m_active_objects.erase(id);
-	delete obj;
+	// Delete the obj before erasing, as the destructor may indirectly access
+	// m_active_objects.
+	it->second.reset();
+	m_active_objects.erase(it);
```

Comment:
couldn't `it` also be invalidated here?

Reply:
Good point, it can (the destructor of luaentity saos uses the metatable of `core.luaentities`).
Fixed.